### PR TITLE
convert RadSuOlsonMG to C++

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2021-01-13-004/index.html

File: Exec/radiation_tests/RadSuOlsonMG/problem_initialize_state_data.H [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    for (int n = 0; n < NumSpec; n++) {
+        state(i,j,k,UFS+n) = 0.0_rt;
+    }
+    state(i,j,k,UFS) = state(i,j,k,URHO);
+
+#if NAUX_NET > 0
+    for (int n = 0; n < NumAux; n++) {
+        state(i,j,k,UFX+n) = state(i,j,k,URHO);
+    }
+    state(i,j,k,UFX) = state(i,j,k,URHO);
```
Comment (harpolea) [inline Exec/radiation_tests/RadSuOlsonMG/problem_initialize_state_data.H:29]: Is this line needed?

**Review by harpolea — CHANGES_REQUESTED** [review]

File: Exec/radiation_tests/RadSuOlsonMG/problem_initialize_state_data.H (hunk 1 above)
Comment (harpolea) [reply Exec/radiation_tests/RadSuOlsonMG/problem_initialize_state_data.H:29]: (line 29)

Comment (zingale) [reply Exec/radiation_tests/RadSuOlsonMG/problem_initialize_state_data.H:29]: I guess not.  There was no indication of what the Aux stuff should be for this problem, but indeed, that line is redundant as written. I'll update the PR.

Comment (zingale) [reply Exec/radiation_tests/RadSuOlsonMG/problem_initialize_state_data.H]: updated

**Review by harpolea — APPROVED** [review]

**Pull request merged**